cpp/CMakeLists.txt

```diff
@@ -175,6 +175,11 @@ target_link_libraries(${PROJECT_NAME}
   ${PROTOBUF_LIBRARIES}
 )
 
+target_compile_options(${PROJECT_NAME}
+  PRIVATE
+    -Wno-error=unusable-partial-specialization
```

Comment:
IMHO, this is a bad idea. I'd prefer to fix the code instead of concealing problems.

Reply:
In order to not conceal the problem, I excluded the new warning from Werror, but left the warning in so it's very obviously still there: https://travis-ci.org/TokTok/jvm-toxcore-c/builds/588156956#L802 (The difference between `-Wno-unusable-partial-specialization` and `-Wno-error=unusable-partial-specialization`.)

If you want to try to fix the warning you're very welcome to, because I had a quick look and won't be able to without reading up on C++ templates.